mlflow/metrics/genai/genai_metric.py

```diff
@@ -245,6 +246,13 @@ def make_genai_metric_from_prompt(
         )
 
     """
+    # When users create a custom metric using this function,the metric configuration
+    # will be serialized and stored. This enables us to later deserialize the configuration,
+    # allowing users to understand their LLM evaluation results more clearly.
+    custom_metric_config = locals()
```

Comment:
If I insert x = 1, would that be included in `custom_metric_config`?

Reply:
yes, it will. and the unit test will fail. that's why I hard coded the expected dict in tests so that it forces whoever breaks the test to manually fix/update it.

could add a comment here to say smth like "DO NOT INSERT THINGS ABOVE" if it can make it more explicit? or is there another easier way to do this in python? using locals cuz I was a bit lazy to manually construct the dict

Reply:
Got it. I'd manually construct the dict, which is more explicit.

Reply:
thought about it and decided to go with adding a comment above instead of manually configure

rationale:
the cons for locals is that people can insert things above which will be included in locals() without them noticing it
but manually constructing the dict will also have a similar problem that people can add new args for the function signature without updating the dict. we almost need to add a comment there if we want to be explicit. so I think they might be similarly implicit in the end